VirtoCommerce.MarketingModule.Data/Services/CouponService.cs

```diff
@@ -51,15 +51,15 @@ public GenericSearchResult<Coupon> SearchCoupons(CouponSearchCriteria criteria)
                 {
                     sortInfos = new[] { new SortInfo { SortColumn = ReflectionUtility.GetPropertyName<Coupon>(x => x.Code), SortDirection = SortDirection.Descending } };
                 }
-                query = query.OrderBySortInfos(sortInfos);
-                
+                query = query.OrderBySortInfos(sortInfos).ThenBy(x => x.Id);
+
                 var searchResult = new GenericSearchResult<Coupon> { TotalCount = query.Count() };
 
                 var ids = query.Select(x => x.Id)
                                .Skip(criteria.Skip)
-                               .Take(criteria.Take).ToArray();
+                               .Take(criteria.Take).ToList();
 
-                searchResult.Results = GetByIds(ids);
+                searchResult.Results = GetByIds(ids.ToArray()).OrderBy(x => ids.IndexOf(x.Id)).ToList(); ;
```

Comment:
Split into multiple lines.
The same for other files.

Reply:
fixed